File: benchmarks/rust/07_string_ops.rs
// String Operations Benchmark - Rust Implementation
// String concatenation and character scanning

fn main() {
    let text = "alpha,beta,gamma,delta;".repeat(400);

    let mut count_a = 0;
    let mut count_comma = 0;
    let mut count_semicolon = 0;

    for ch in text.chars() {
        if ch == 'a' {
            count_a += 1;
        } else if ch == ',' {
            count_comma += 1;
        } else if ch == ';' {
            count_semicolon += 1;
        }
    }

    let checksum = text.len() as i64 + count_a + count_comma + count_semicolon;
    println!("string operations checksum: {}", checksum);
}

File: benchmarks/python/07_string_ops.py
def main():
    text = "alpha,beta,gamma,delta;" * 400

    count_a = 0
    count_comma = 0
    count_semicolon = 0
    for ch in text:
        if ch == "a":
            count_a += 1
        elif ch == ",":
            count_comma += 1
        elif ch == ";":
            count_semicolon += 1

    print(len(text) + count_a + count_comma + count_semicolon)


if __name__ == "__main__":
    main()
